src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -1810,6 +1810,77 @@ internal MethodSymbol EnsureThrowIfNullFunctionExists(SyntaxNode syntaxNode, Syn
             Debug.Assert((object?)privateImplClass.GetMethod(PrivateImplementationDetails.SynthesizedThrowFunctionName)!.GetInternalSymbol() == internalSymbol.ThrowMethod);
             return internalSymbol;
         }
+
+        /// <summary>
+        /// Creates the ThrowSwitchExpressionException helper if needed.
+        /// </summary>
+        internal MethodSymbol EnsureThrowSwitchExpressionExceptionExists(SyntaxNode syntaxNode, SyntheticBoundNodeFactory factory, DiagnosticBag? diagnostics)
```

Comment:
nit: I couldn't find a test for `VerifyIL` on `ThrowSwitchExpressionException`. Consider adding.
Also, it would be good to ensure we have tests executing those three private implementation methods (`CompileAndVerify` with `expectedOutput:`).

Reply:
Will update `UnmatchedInput_04` to verify IL.

`UnmatchedInput_04`, `UnmatchedInput_02`, and `UnmatchedInput_01` all execute the synthesized methods using `CompileAndVerify`.